### DISCUSSION TITLE
Gitlab pages instructions for Pelican needs work

### DISCUSSION THREAD

**Discussion by damascene** [getpelican/pelican#3289] [Q&A]
I'm trying to use Pelican for my new project and things are working fine for the most part however I'm having issues so I decided to do some testing using [Pelican template from Gitlab](https://gitlab.com/pages/pelican/). I discovered few issues.

### Local Version
Local version of the Pelican quick-start shows a  nice version:

![image](https://github.com/getpelican/pelican/assets/411046/2f801978-b074-4bb3-a282-63214f0c27d4)

### Gitlab Version

The version from gitlab after following instruction results in this:

![image](https://github.com/getpelican/pelican/assets/411046/47d00d5d-22ef-4cc5-886d-3bf60716c93d)

Page live at:
https://testing_pelican.gitlab.io/

Source code:
https://gitlab.com/testing_pelican/testing_pelican.gitlab.io

## backports-zoneinfo Issue

Before building I've fixed a `backports-zoneinfo` issue:

```
ERROR: Could not build wheels for backports-zoneinfo, which is required to install pyproject.toml-based projects
```

By increasing version of Alpine to:
```
image: python:3.9-alpine
```

## Broken links 

Links on the page are broken it directs to:

```
https://testing_pelican.gitlab.io/pelican/pelican-on-gitlab-pages.html
```

instead of:
```
https://testing_pelican.gitlab.io/pelican-on-gitlab-pages.html
```

## Group or Username Site

Instruction doesn't explicitly say that you have to create the repository with name `your_group.gitlab.io` but it says you should rename. However renaming would keep the directory name intact and it that will have you visit the domain with subdirectory. 

## Allow Public Access to Pages

On Gitlab/pages you will notice a notification to allow public access in Giltab Settings/Visibility 

## Comment 

I'm already running few pelican static websites and I've managed to use different theme but the default installation should at least show some beautiful working example.

**Comment by justinmayer** [comment]
Hi @damascene. Folks from the community kindly added support for Pelican to GitLab Pages, and I don't believe any of the core Pelican maintainers (myself included) have ever used said GitLab Pages integration. Perhaps @axilleas can chime in with ideas regarding the best place to report issues related to this integration effort?

**Reply by axilleas** [reply]
@damascene yep, like Justin suggested, please open an issue at https://gitlab.com/pages/pelican/-/issues.

I was in the process of updating the CI config, but haven't finished yet, I will try to pick it up next week.

I'll make sure to add a note in the readme to open issues in the GitLab issue tracker where the template resides.

> However renaming would keep the directory name intact and it that will have you visit the domain with subdirectory.

We must rephrase that. You should really [change the path](https://docs.gitlab.com/ee/user/project/working_with_projects.html#rename-a-repository) of the project from its settings, not the name.
